docs/source/posts/2021/2021-06-21-week-3-antriksh.rst

```diff
@@ -0,0 +1,31 @@
+Week #3: Adapting GridLayout to work with UI
+============================================
+
+.. post:: June 21 2021
+   :author: Antriksh Misri
+   :tags: google
+   :category: gsoc
+
+What did I do this week?
+------------------------
+This week my tasks revolved around layout and UI elements. The primary goal for this week was to adapt the GridLayout to work with different UI elements. Currently, GridLayout just supports vtk actors and not UI elements, my task was to modify the class to support UI elements. The other tasks for this week are described below in detail:-
+
+* `Adapt GridLayout to support UI elements <https://github.com/fury-gl/fury/pull/443>`_ : This was the main task for the week and the aim for this was to actually modify GridLayout to support UI elements. This was not possible before because GridLayout only supported vtk actors (because of certain methods only being provided by vtk actors). I modified the main class itself along with some utility functions. The problem that I faced during this was circular imports. Currently, the structure of FURY doesn't allow certain modules to be imported into other modules because of circular imports. A way to get around this was to actually import the modules inside the methods but this is not ideal always. This will be fixed in the future PRs where the UI module will be redesigned. I also added support for grid position offsetting, which basically means that the position of the UI elements that are placed in the Grid can be offset by a global offset passed in the constructor of GridLayout class. Below is an example showing the current state of GridLayout with different UI elements:- . I also created a brief example to demonstrate how to use GridLayout of different cellshapes with UI elements link to which is `here <https://github.com/fury-gl/fury/pull/443/files#diff-853d17c3134e7d22de88523bb787dc05d52ec798dc2111aa0419dfd5d634350a>`_
+
+    .. image:: https://i.imgur.com/EX2cN1i.png
```

Comment:
How long this image will stay there? you might need to save the image somewhere else, like a branch in your repo.

Reply:
The image is hosted on imgur which I think keep it there for as long as you want.